toplevel/coqargs.ml

```diff
@@ -508,6 +510,25 @@ let parse_args arglist : coq_cmdopts * string list =
 
     |"-o" -> { oval with compilation_output_name = Some (next()) }
 
+    |"-bytecode-compiler" ->
+      { oval with enable_VM = get_bool opt (next ()) }
+
+    |"-native-compiler" ->
+
+      (* We use two boolean flags because the four states make sense, even if
+      only three are accesible to the user at the moment. The selection of the
+      produced artifact(s) (`.vo`, `.vio`, `.coq-native`, ...) should be done by
+      a separate flag, and the "ondemand" value removed. *)
+      let (enable,precompile) =
+        match (next ()) with
+        | ("yes" | "on") -> true, true
+        | "ondemand" -> true, false
+        | ("no" | "off") -> false, false
```

Comment:
I don't see how to factor cleanly, since `get_bool` errors on "ondemand". I can move it above, as you suggest. Note that, as mentioned in the comment, "ondemand" is meant to disappear soon.

Reply:
Then add a comment saying to use get_bool when ondemand is removed.

Reply:
done